modules/dnn/src/layers/prior_box_layer.cpp

```diff
@@ -280,6 +277,14 @@ class PriorBoxLayerImpl : public PriorBoxLayer
     }
 
 #ifdef HAVE_OPENCL
+    static inline void setKernelArg(ocl::Kernel& kernel, int id, const UMat& m)
+    {
+        if (!m.empty())
+            kernel.set(id, ocl::KernelArg::PtrReadOnly(m));
+        else
+            kernel.set(id, nullptr);
```

Comment:
I believe we should avoid this approach.
It is better to use compile-time definitions and `#ifdef` blocks in OpenCL code instead of dummy parameters. It will produce more effective code without such hacks.

Reply:
@alalek, thanks! I'll manage it.